2024.10 (#35016)
home-assistant/home-assistant.github.io: source/_posts/2024-10-02-release-202410.markdown

- **Support for operation states**  
  Control and monitor the operation state of, for example, dishwashers
  and other household equipment.
- **Support for smoke and CO sensors**  
  Monitor the air quality in your home with smoke and CO sensors.
- **Add support for Matter 1.3 power/energy sensors**  
  Especially the energy sensors are great to use with the energy dashboard.

[Matter integration]: /integrations/matter

## Other noteworthy changes

There are many more improvements in this release; here are some of the other
noteworthy changes this release:

- In June, we added support for adding [Model IDs to devices]. This release
  has lots of integrations that add them, including [Matter], [Plugwise],
  [Wallbox], and [Govee]. We are calling for contributions to add model IDs to
  more integrations! Maybe a cool suggestion for an [Hacktoberfest] contribution!
- [@bmesuere] improved the accessibility of all graphs by adjusting all the
  default colors used in those graphs! You are a hero!
- When you unignore an integration or device on the integrations page,
  Home Assistant will instantly try to rediscover it again! Nice! Thanks,
  [@emontnemery]!

> - We now have a new unit of measurement for energy entities: calories. Useful,

Reviewer: ```suggestion
- We now have a new unit of measurement for energy entities: calories. Useful
```

[@wendevlin]: https://github.com/wendevlin

### YAML-editor improvements

[@naps62] has been busy improving the YAML editor found in the Home Assistant
user interface.

We now have a vertical line based on indentation, which makes it easier to see
the indentation of your YAML configuration and allows you to spot and prevent
indentation errors.

<img class="no-shadow" src="/images/blog/2024-10/YAML-editor-improvements.png" alt="Screenshot showing the YAML editor with collapsed sections and the vertical indentation line.">

But he didn’t stop there. You can now collapse and expand sections of your
YAML configuration based on the indentation level; which is also demonstrated
in the screenshot above.

Nice work [@naps62]!

[@naps62]: https://github.com/naps62

## Need help? Join the community!

Home Assistant has a great community of users who are all more than willing
to help each other out. So, join us!

Our very active [Discord chat server](/join-chat) is an excellent place to be
at, and don't forget to join our amazing [forums](https://community.home-assistant.io/).

> Found a bug or issue? Please report it in our [issue tracker](https://github.com/home-assistant/core/issues),

Reviewer: ```suggestion
Found a bug or issue? Please report it in our [issue tracker](https://github.com/home-assistant/core/issues)
```

called the heading card. Making them a card rather than an inline title gives
you a lot more flexibility, allowing you to put them nearly anywhere.

The heading card can be used to make titles, but also subtitles, which are half
the height and have smaller, lighter text. So, if you have ever had a section
that needs to be subdivided, slap a subtitle in there.

![Screenshot showing a dashboard using heading cards with titles and subtitles](/images/blog/2024-10/headings.png)

Titles and subtitles can be made clickable, displaying a chevron
{% icon "mdi:chevron-right" %} when you can interact with them. This can
perform an action or navigate to a subpage, like having an energy section title
navigate to the full energy dashboard.

Best of all, if you've already added titles to your sections, they will be
migrated automatically to the new card. 😎

[Sections]: /dashboards/sections/

### Another headline feature

We had such a fantastic reaction when we released our [renewed badges],
and this new heading card has something very similar and equally as visually
appealing. It can show multiple entity states to the right of the title or
subtitle.

> ![Screenshot showing a dashboard with additional entities show on card headings](/images/blog/2024-10/heading-with-entities.png)

Reviewer 2: ```suggestion
![Screenshot showing a dashboard with additional entities with on card headings](/images/blog/2024-10/heading-with-entities.png)
```

colder this year compared to last.

![Screenshot showing a long-term statistic repair showing up in the settings menu](/images/blog/2024-10/statistics-repair.jpg)

In the past, if a statistic encountered an issue, it might be quietly broken
until you dig into the statistics page. Now, Home Assistant will raise a repair
notification when it detects issues with a long-term statistic and give
you options for fixing the issue.

[your energy usage]: /docs/energy/
[Long-term statistics]: https://data.home-assistant.io/docs/statistics/

## Improved YAML syntax for automations

Recently, we renamed `service` to `action` in the [Home Assistant UI and YAML].
This was done with the goal of making the syntax more natural allowing it
to be easier to grasp and read. We want to continue this work to improve the
YAML syntax, this time focusing on {% term automations %}, and we've made the
following changes,

- The top-level `trigger` key is now `triggers` (plural)
- The top-level `condition` key is now `conditions` (plural)
- The top-level `action` key is now `actions` (plural)
- The `platform` key of a trigger definition is now `trigger`

> The result of all this is YAML automation syntax now has `triggers` with

Reviewer 2: ```suggestion
The result of all this is that YAML automation syntax now has `triggers` with
```

multiple `trigger` definitions, `conditions` with multiple `condition`
definitions, and `actions` with multiple `action` definitions.

> Here is a small example of an automation that highlighting these changes.

Reviewer 2: ```suggestion
Here is a small example of an automation highlighting these changes.
```

<p class="img">

> <img src="/images/blog/2024-10/YAML-automation-syntax-changes.png" alt="Screenshots; left showing the old automation YAML style, on the right shows the new style. The differences between them have been highlighted.">

Reviewer 2: ```suggestion
<img src="/images/blog/2024-10/YAML-automation-syntax-changes.png" alt="Screenshots; left showing the old automation YAML style, right showing the new style. The differences between them have been highlighted.">
```

Author: That won't work in this case, due to the subtext

Reviewer 2: I'm not following.

Author: Who VScode showed something else; something goes shout with the rendering in there... will adjust 👍

The 2024.9 release on the left and this 2024.10 release on the right.
</p>

As with the other syntax changes we've done recently, this is
**not a breaking change**, and there will be no deprecation. The previous syntax
will continue to work, and there are no plans to remove the old syntax. The new
syntax is recommended, and all documentation and examples have been updated.
Automations managed by the automation editor will automatically be migrated to
the new syntax when saving an automation.

This is not the only change to our syntax, thanks to [@karwosts] we now support
[merging a list of triggers], which is mostly useful for

> {% term blueprint blueprint %} authors.

Reviewer 2: ```suggestion
{% term blueprint %} authors.
```

fully transparent to the user, and no action is required. Just sit down, relax,
and enjoy the faster upgrade process!

[@edenhaus]: https://github.com/edenhaus
[developer backstory]: https://developers.home-assistant.io/blog/2024/04/03/build-images-with-uv/
[opt-in analytics]: https://analytics.home-assistant.io/
[speed boost for downloads]: /blog/2024/08/21/hacs-the-best-way-to-share-community-made-projects/#faster-downloads

## Integrations

Thanks to our community for keeping pace with the new {% term integrations %}
and improvements to existing ones! You’re all awesome 🥰

### New integrations

We welcome the following new integrations in this release:

- **[Cambridge Audio]**, added by [@noahhusby]  
  Control all receivers and streamers that support the [StreamMagic app].
- **[Deako Smart Lighting]**, added by [@Balake]  
  Control and automate your [Deako] devices.
- **[Duke Energy]**, added by [@hunterjm]  
  Import your energy usage data from Duke Energy into the Home Assistant energy
  dashboard.
- **[Google Photos]**, added by [@allenporter]  
  Upload photos to your [Google Photos library] and use as a [media source].

>   Note that only images uploaded by the integration can use viewed and used.

Reviewer 2: ```suggestion
  Note that only images uploaded by the integration can be viewed and used.
```

- **[Iskra]**, added by [@iskrakranj]  

>   Monitor your energy and power usage from a Iskra energy meter.

Reviewer 2: ```suggestion
  Monitor your energy and power usage from an Iskra energy meter.
```

In terms of stability, we have improved the robustness of the integration to
ensure it sets up correctly every time, even if an issue with one
of your devices occurs. Besides the usual bugfixes and improvements, this
release has some cool new features and additions:

- **Button entities**  
  New button entities that allow you to perform an action on a device. For
  example, to identify a device or reset the filter state.
- **Valve entities**  
  Support for water valves, allowing you to control the flow of fluids or gases.
- **Support for operation states**  
  Control and monitor the operation state of, for example, Dishwashers
  and other household equipment.
- **Support for Smoke and CO sensors**  
  Monitor the air quality in your home with smoke and CO sensors.
- **Add support for Matter 1.3 power/energy sensors**  
  Especially the energy sensors are great to use with the energy dashboard.

[Matter integration]: /integrations/matter

## Other noteworthy changes

There are many more improvements in this release; here are some of the other
noteworthy changes this release:

> - In June, we added support for adding [Model IDs to devices], this release

Reviewer 2: ```suggestion
- In June, we added support for adding [Model IDs to devices]. This release
```

>   has lots of integrations that add them. Including [Matter], [Plugwise],

Reviewer 2: ```suggestion
  has lots of integrations that add them, including [Matter], [Plugwise],
```

>   [Wallbox], and [Govee]. We are calling contributions to add model IDs to

Reviewer 2: ```suggestion
  [Wallbox], and [Govee]. We are calling for contributions to add model IDs to
```

  more integrations! Maybe a cool suggestion for an [Hacktoberfest] contribution!

> - When you unignore an or integration or device on the integrations page,

Reviewer 2: ```suggestion
- When you unignore an integration or device on the integrations page,
```

  Home Assistant will instantly try to rediscover it again! Nice! Thanks,
  [@emontnemery]!
- We now have a new unit of measurement for energy entities: calories. Useful,

>   for like devices like a treadmill or an exercise bike. Thanks, [@dudanov]!

Reviewer 2: ```suggestion
  for devices like a treadmill or an exercise bike. Thanks, [@dudanov]!
```

- [@TrevorSchirmer] did something similar, and added a new unit of speed:
  mm/s and in/s. Thank you!
- When you rename a device, you will now see more details about the rename
  operation. Including a preview of the new entity IDs. Nice, [@karwosts]!
- [@gjohansson-ST] added a new template function [`merge_response`] that

>   allows you to merge multiple response of single or multiple actions into

Reviewer 2: ```suggestion
  allows you to merge multiple responses of a single action or multiple actions into
```

>   a single response. This is useful when you want to combine the responses

Reviewer 2: I don't think this sentence adds anything. It repeats the previous sentence.
```suggestion
  a single response.
```

>   of multiple services into a single response. Nice!

Reviewer 2: ```suggestion
 Nice!
```

- [Trigger-based templates] now have support for adding conditions! Just like
  automations, you can now add conditions to these entities to control when
  they are rendered. Thanks, [@chammp]! 
- More template fun! You can now set up your templated alarm control panel
  straight from the UI. Thanks, [@gjohansson-ST]!

> - The statistics developer tools have been improved. They now has the new-style

Reviewer 2: ```suggestion
- The statistics developer tools have been improved. They now have the new-style
```

[@karwosts]: https://github.com/karwosts
[@TrevorSchirmer]: https://github.com/TrevorSchirmer
[`merge_response`]: /docs/configuration/templating/#merge-action-responses
[Assist]: /voice_control
[Govee]: /integrations/govee_light_local
[Hacktoberfest]: https://hacktoberfest.digitalocean.com/
[Matter]: /integrations/matter
[Model IDs to devices]: https://developers.home-assistant.io/blog/2024/07/16/device-info-model-id
[Plugwise]: /integrations/plugwise
[Trigger-based templates]: /integrations/template/#trigger-based-template-binary-sensors-buttons-images-numbers-selects-and-sensors
[Wallbox]: /integrations/wallbox

### The source of a repair

{% term Integrations %} can raise issues in your repairs dashboard, they can
raise issues for themselves, but also for other integrations. An example of
such an issue, is an integration that connects to your device that raises an
issue on an {% term automation %} that controls it.

The result, is an automation issue in your repair center. This is cool, but
you might want to know which integration raised the issue in the first place.

Thanks to [@wendevlin], we now show the source integration of the repair issue
in the issue dialog.

> <img class="no-shadow" src="/images/blog/2024-10/repair-issue-reported-by.png" alt="Screenshot showing the YAML editor with collapsed sections and the vertical indentation line.">

Reviewer 2: The alt text is copy paste from the YAML-editor improvements below. Needs to be corrected.

provided climate entity.

([@gjohansson-ST] - [#125246]) ([documentation](/integrations/ecobee))

[@gjohansson-ST]: https://github.com/gjohansson-ST
[#125246]: https://github.com/home-assistant/core/pull/125246

{% enddetails %}

{% details "Honeywell Total Connect Comfort (US)" %}

The previously deprecated `aux_heat` attribute has been removed from the
provided climate entity.

([@gjohansson-ST] - [#125248]) ([documentation](/integrations/honeywell))

[@gjohansson-ST]: https://github.com/gjohansson-ST
[#125248]: https://github.com/home-assistant/core/pull/125248

{% enddetails %}

{% details "Humidifier" %}

We have added more validation to the `humidifier.set_humidity` action, which
enforces that the target humidity is within the targeted entity's minimum
and maximum humidity levels.

You might need to adjust automations and scripts to use valid values if you
are using this action.

([@gjohansson-ST] - [#125863]) ([documentation](/integrations/humidifier))

[@gjohansson-ST]: https://github.com/gjohansson-ST
[#125863]: https://github.com/home-assistant/core/pull/125863

{% enddetails %}

{% details "Nexia/American Standard/Trane" %}

> The `aux_heat` attribute of the climate entity is now deprecated and will be

Reviewer 2: This is a deprecation. Maybe ok to mention anyway, but I'm not sure it's intended.

Author: You are right, this will raise a repair
